### PR TITLE
Feat: add meilisearch index actions with version compactible.

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
- [x] update the action to scrap whenever the docs changes using pymeilisearch
- [x] update the index according to version

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import numpy as np
 import json
 from sphinx_gallery.sorting import FileNameSortKey
-from ansys_sphinx_theme import ansys_favicon, get_version_match, pyansys_logo_black
+from ansys_sphinx_theme import ansys_favicon, get_version_match, pyansys_logo_black, con
```
Comment (Revathyvenugopal162) [inline doc/source/conf.py:14]:
```suggestion
from ansys_sphinx_theme import ansys_favicon, get_version_match, pyansys_logo_black
```

File: .github/workflows/full_documentation.yml [hunk 2]
```diff
@@ -89,3 +90,56 @@ jobs:
           cname: ${{ env.DOCUMENTATION_CNAME }}
           token: ${{ secrets.GITHUB_TOKEN }}
           python-version: ${{ matrix.python-version }}
+
+  
+  doc-index-stable:
```
Comment (jorgepiloto) [inline .github/workflows/full_documentation.yml:95]: Let us parametrize this job as two different API references are required. Also, make migrate this to use ansys/actions.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]

File: .github/workflows/full_documentation.yml (hunk 2 above)
Comment (MaxJPRey) [reply .github/workflows/full_documentation.yml:95]: That would be great if we can reuse generic actions.

File: .github/workflows/full_documentation.yml [hunk 3]
```diff
@@ -89,3 +90,54 @@ jobs:
...
+          echo "VERSION_MEILI=$VERSION_MEILI" >> $GITHUB_ENV
+
+      - name: "Deploy the stable documentation index for PyAEDT API"
+        uses: ansys/actions/doc-deploy-index@v4
+        with:
+          cname: ${{ env.DOCUMENTATION_CNAME }}/version/${{ env.VERSION }}
+          index-name: pyaedt-v${{ env.VERSION_MEILI }}
+          host-url: ${{ vars.MEILISEARCH_HOST_URL }}
+          api-key: ${{ env.MEILISEARCH_API_KEY }}
+          pymeilisearchopts: --stop_urls \"EDBAPI\"
```
Comment (MaxJPRey) [inline .github/workflows/full_documentation.yml:133]: Maybe add a comment here for later maintenance.
Some people might not be familiar with those pymeiliserach options and the consequences on EDB API.

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Comment by Revathyvenugopal162** [comment]
Thank you for the review @MaxJPRey  and @jorgepiloto , merging!

**Pull request merged**